### DISCUSSION TITLE
TypeError: vertojs__WEBPACK_IMPORTED_MODULE_1__.Verto is not a constructor

### DISCUSSION THREAD

**Discussion by gregoriusus** [roman-yerin/vertojs#24] [Q&A] [answered]
If I use library from src everything is working, but if I use it from  dist, I get this error... Can you please advise?

**Comment by roman-yerin** [comment]
Can you provide me with a code you imported a library at?

**Comment by gregoriusus** [comment]
I installed library via npm.

This is how I import it in code:
//import { Verto } from "../../../../../node_modules/vertojs/dist/index";
import {Verto} from 'vertojs';

Commented line is working, regular one doesn't. And it fails when initializing Verto:
var verto = new Verto({
      transportConfig:{
        login: "xxx",
        passwd: "xxx",
        socketUrl: "xxx"
      },rtcConfig:{
        iceServers: [{urls:"stun:stun.l.google.com:19302"}]
      },
      debug: true,
      ice_timeout: 5000
    })

**Comment by roman-yerin** [comment]
Well... Do you have typescript support on the project? The default package entry point assumes you have. I will review the library code and defaults next days to make it more compatible to default user expectations.

**Comment by gregoriusus** [comment]
Thank you for being active on this project. I am developing app in Angular, so typescript is supported.

**Comment by gregoriusus** [comment]
I uninstalled npm and reinstall it. Now I do not have logout() and isLogged() functions in src/index.ts. I tried to download npm and open compressed file and this piece of code has gone. Something changed in npm?

**Comment by roman-yerin** [comment, marked as answer]
Nothing changed when you asked.
I have just rebuilt the library, now you should be able to use it as described
import { Verto } from 'vertojs'

**Comment by gregoriusus** [comment]
It does work now 👍  Thank you

**Comment by AhreemaSheikh** [comment]
Hi, @gregoriusus can you please share the version of npm and node you are currently using?

**Reply by gregoriusus** [reply]
Here:
npm: '8.1.2',
node: '16.13.1'
Are you using this library?

**Comment by AhreemaSheikh** [comment]
Thank you @gregoriusus. I am using this library and getting the same error 
`var verto_fs = new verto(verto); TypeError: verto is not a constructor`

**Reply by roman-yerin** [reply]
Hi, it depends on how have you imported the library.
If you use import directive, like `import { Verto } from 'vertojs'`, you have to change the first letter in the constructor name to be capital `var verto_fs = new Verto(verto);`

**Reply by AhreemaSheikh** [reply]
Thank you so much for taking out time to help me. 
But I really don't understand it.
I have done the same as you said, I get the following error. 
![vertojs](https://user-images.githubusercontent.com/114995340/201585817-f5876e5b-7066-4ec3-a446-9b59e5d1daab.png)
![node](https://user-images.githubusercontent.com/114995340/201585857-08a34b5e-ca42-44a8-b4ce-d4a63468559c.png)

**Reply by Kiddo-middo** [reply]
@AhreemaSheikh I'd love to tell, but Idk

**Comment by gregoriusus** [comment]
Try import with brackets

On Mon, 14 Nov 2022 at 06:59, AhreemaSheikh ***@***.***>
wrote:

> Thank you so much for taking out time to help me.
> I don't really understand it.
> I have done the same as you said, I get the following error.
> [image: vertojs]
> <https://user-images.githubusercontent.com/114995340/201585817-f5876e5b-7066-4ec3-a446-9b59e5d1daab.png>
> [image: node]
> <https://user-images.githubusercontent.com/114995340/201585857-08a34b5e-ca42-44a8-b4ce-d4a63468559c.png>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/roman-yerin/vertojs/discussions/24#discussioncomment-4131952>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABAR5XYABZMYMIIHDUCYKIDWIHIMXANCNFSM5LGBTBPA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Gregor Nanger

*CTO*
t./f.: 00386 (0) 7 6000 308/309 • m:. 00386 (0)41 756485
• Infomedia d.o.o. • Jerebova 3, Novo mesto, Slovenia
• www.infomedia.si

**Reply by roman-yerin** [reply]
Yeah, import { Verto } and don't pass undefined variable verto to the constructor.

<pre>
import { Verto } from 'vertojs'
let verto = new Verto(options) // if you are ok with the defaults, call new Verto()
</pre>
